meshio/_files.py

```diff
@@ -0,0 +1,22 @@
+import os
+import sys
+from contextlib import contextmanager
+
+
+def is_buffer(obj, mode):
+    return ("r" in mode and hasattr(obj, "read")) or (
+        "w" in mode and hasattr(obj, "write")
+    )
+
+
+@contextmanager
+def open_file(path_or_buf, mode="r"):
+    if is_buffer(path_or_buf, mode):
+        yield path_or_buf
+    elif sys.version_info < (3, 6) and isinstance(path_or_buf, os.PathLike):
+        # todo: remove when 3.5 is EoL
+        with open(str(path_or_buf), mode) as f:
+            yield f
+    else:
+        with open(path_or_buf, mode) as f:
```

Comment:
Not `yield open(path_or_buf, mode)`? (Not familar with `@contextmanager`. I thought its purpose was that you can use the class with `with`.)

Reply:
`@contextmanager` means that code before the `yield` is run on `__enter__`, then code after the `yield` is run on `__exit__`. Wrapping the `yield`s in `with` statements means that on `__exit__`, the file's `__exit__` is also (semi-explicitly) run, where if you just yield the result, then you're relying on whatever it is that closes file handles when they go out of scope, which I don't trust as much.

Reply:
Got it.